### ISSUE TITLE
Histogram Autolevels does not work for constant arrays

### ISSUE THREAD

**Issue by kremeyer** [pyqtgraph/pyqtgraph#2206]
### Short description
When continuously plotting images that is constant (all values are the same) to an ImageView widget with the automatic levels for the histogram enables, the axis range will keep extending infinitely in both directions around that constant.

### Code to reproduce
In this example you can just click the `Norm` button over and over again to see the behavior.

```python
import pyqtgraph as pg
import numpy as np
from pyqtgraph.Qt import QtWidgets

app = pg.mkQApp('Auto Level Bug')
win = QtWidgets.QMainWindow()
win.resize(800,800)
imv = pg.ImageView()
win.setCentralWidget(imv)
win.show()

def new_img(self):
	imv.setImage(np.zeros((100, 100)))
	print('new img')
imv.__dict__['new_img'] = new_img
imv.ui.menuBtn.clicked.connect(new_img)

if __name__ == '__main__':
    pg.exec()
```

### Expected behavior
Axis bounds should not change.

### Real behavior
Auto range keeps extending axis limits of histogram towards `(-inf,inf)` with every consecutive image drawn

### Tested environment(s)

 * PyQtGraph version: 0.12.3
 * Qt Python binding: PyQt5 5.15.4 Qt 5.15.2
 * Python version: 3.8.10
 * NumPy version: 1.21.2
 * Operating system: Ubuntu & Windows 10
 * Installation method: pip

**Comment by lyragordon** [comment]
Having a similar problem. With any data outside (0,1), it will just peak the colormap. Reproduced with a minor change to @kremeyer 's code (Norm button produces the behavior I'd expect from autolevel):
```
import pyqtgraph as pg
import numpy as np
from pyqtgraph.Qt import QtWidgets

app = pg.mkQApp('Auto Level Bug')
win = QtWidgets.QMainWindow()
win.resize(800, 800)
imv = pg.ImageView()
win.setCentralWidget(imv)
win.show()


def new_img(self):
    imv.setImage(np.random.rand(100, 100) * 10, autoLevels=True)
    print('new img')


imv.__dict__['new_img'] = new_img
imv.ui.menuBtn.clicked.connect(new_img)

if __name__ == '__main__':
    pg.exec()

```

**Comment by j9ac9k** [comment]
Fixed by merging #2489
